From 31a1519c8f8cdaf768485cf9b7f7cac67d533b92 Mon Sep 17 00:00:00 2001
Message-Id: <31a1519c8f8cdaf768485cf9b7f7cac67d533b92.1413837857.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: vkorjani <vikas.korjani@intel.com>
Date: Thu, 27 Mar 2014 09:58:09 +0530
Subject: [PATCH 209/312] FOR_UPSTREAM [VPG]: drm/i915: Parsing Backlight-ON,
 Backlight-OFF, and Tear-On sequence.

New sequence are added in GOP to support Backlight enabling
and Disabling. also new sequence element I2C is been added
this patch provide support to parse thse sequences in driver.

Change-Id: I3c1f184dd641713fda323aa2360cb5c043cd0af1
Signed-off-by: vkorjani <vikas.korjani@intel.com>
---
 drivers/gpu/drm/i915/intel_bios.c |    8 +++++++-
 drivers/gpu/drm/i915/intel_bios.h |    4 ++++
 2 files changed, 11 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/intel_bios.c b/drivers/gpu/drm/i915/intel_bios.c
index ad15b4f..516710f 100644
--- a/drivers/gpu/drm/i915/intel_bios.c
+++ b/drivers/gpu/drm/i915/intel_bios.c
@@ -723,6 +723,12 @@ static u8 *goto_next_sequence(u8 *data, int *size)
 
 			data += 3;
 			break;
+		case MIPI_SEQ_ELEM_I2C:
+			/* skip by this element payload size */
+			data += 6;
+			len = *data;
+			data += len + 1;
+			break;
 		default:
 			DRM_ERROR("Unknown element\n");
 			return NULL;
@@ -867,7 +873,7 @@ parse_mipi(struct drm_i915_private *dev_priv, struct bdb_header *bdb)
 			dev_priv->vbt.dsi.sequence[seq_id] = data;
 			DRM_DEBUG_DRIVER("Found mipi sequence - %d\n", seq_id);
 		} else {
-			DRM_ERROR("undefined sequence\n");
+			DRM_ERROR("undefined sequence - %d\n", seq_id);
 			goto err;
 		}
 
diff --git a/drivers/gpu/drm/i915/intel_bios.h b/drivers/gpu/drm/i915/intel_bios.h
index c56553f..b722e53 100644
--- a/drivers/gpu/drm/i915/intel_bios.h
+++ b/drivers/gpu/drm/i915/intel_bios.h
@@ -923,6 +923,9 @@ enum mipi_seq {
 	MIPI_SEQ_DISPLAY_ON,
 	MIPI_SEQ_DISPLAY_OFF,
 	MIPI_SEQ_DEASSERT_RESET,
+	MIPI_SEQ_BACKLIGHT_ON,
+	MIPI_SEQ_BACKLIGHT_OFF,
+	MIPI_SEQ_TEAR_ON,
 	MIPI_SEQ_MAX
 };
 
@@ -931,6 +934,7 @@ enum mipi_seq_element {
 	MIPI_SEQ_ELEM_SEND_PKT,
 	MIPI_SEQ_ELEM_DELAY,
 	MIPI_SEQ_ELEM_GPIO,
+	MIPI_SEQ_ELEM_I2C,
 	MIPI_SEQ_ELEM_STATUS,
 	MIPI_SEQ_ELEM_MAX
 };
-- 
1.7.9.5

